space-concordia-robotics/robotics-prototype
# Ar tag detection 481

# Assignee Section

## Description
This pull request integrates the ar tag detection from [here](https://github.com/space-concordia-robotics/docker-erc) to this repository.


#### Prerequisites
- Make sure you install [ar_track_alvar](http://wiki.ros.org/ar_track_alvar): `sudo apt install ros-melodic-ar-track-alvar`
- Calibrate your camera by following the instructions [here](http://wiki.ros.org/camera_calibration/Tutorials/MonocularCalibration)
- Print out some AR tags. They can be found [here](http://wiki.ros.org/ar_track_alvar?action=AttachFile&do=view&target=markers0to8.png) for example.


### Steps for Testing
1. Run the project and visit `http://localhost:5000`.
2. Select a camera stream and start it.
3. Click on the "AR" button, you should see a red square appear.
4. Hold up the printed AR tags in front of you camera. The red square should turn green.
5. Click on the "AR" button again and the square should disappear.

#### Additional Steps
1. Test with multiple streams simultaneously.


closes #481

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
